tools/encoder/src/TMXTypes.h

```diff
@@ -0,0 +1,252 @@
+#ifndef TMXTYPES_H_
+#define TMXTYPES_H_
+
+#include <vector>
+#include <set>
+#include <map>
+#include <string>
+
+#include "Vector2.h"
+
+namespace gameplay
+{
+
+typedef std::map<std::string, std::string> TMXProperties;
+
+/**
+ * Represents the tiles that make up a map.
+ */
+class TMXTileSet
+{
+public:
+    /**
+     * Constructor.
+     */
+    TMXTileSet();
+
+    void setImagePath(const std::string& path);
+    const std::string& getImagePath() const;
+
+    void setFirstGid(unsigned int gid);
+    unsigned int getFirstGid() const;
+
+    void setMaxTileWidth(unsigned int value, bool adjustTiles = true);
+    unsigned int getMaxTileWidth() const;
+	void setMaxTileHeight(unsigned int value, bool adjustTiles = true);
+    unsigned int getMaxTileHeight() const;
+
+	void setSpacing(unsigned int value, bool adjustTiles = true);
+    unsigned int getSpacing() const;
+	void setMargin(unsigned int value, bool adjustTiles = true);
+    unsigned int getMargin() const;
+
+    // This will calculate the number of tiles based on max tile size, spacing, and margin.
+    // If any of those change, this needs to be recalled.
+    void setImageWidth(unsigned int value);
+	void setImageHeight(unsigned int value);
+    unsigned int getHorizontalTileCount() const;
+	unsigned int getVerticalTileCount() const;
+
+    void setOffset(const Vector2& offset);
+    const Vector2& getOffset() const;
+
+	static unsigned int calculateImageDimension(unsigned int tileCount, unsigned int tileSize, unsigned int spacing = 0, unsigned int margin = 0);
+	static Vector2 calculateTileOrigin(const Vector2& pos, const Vector2& tileSize, unsigned int spacing = 0, unsigned int margin = 0);
+
+private:
+    std::string _path;
+
+	unsigned int _imgWidth;
```

Comment:
I will take a look at it soon. I thought it was all good, and converted tabs to spaces, so I didn't expect it would have formatting issues.